### PR TITLE
EIP 3549: Metadata Signature Extension 

### PR DESCRIPTION
**Pull request by claytantor**
## Simple Summary
Add signing to the metadata standard and required fields.

## Abstract

Recently Tim Schneider, from Artnet posted a great story about how a whitehat hacker managed to trick the blockchain into forging tokens that would forge the Provenance of the originator of the NFT to look as though it was the originating artist.

[NFT Theft](https://news.artnet.com/opinion/sleepminting-nftheft-monsieur-personne-1960744)

I have added a proposal outlined in the article [Solving For Sleepminting](https://claytantor.wordpress.com/2021/04/22/solving-for-sleepminting/)

* Add md5 hash to metadata as verifiable
* Metadata has must be unique for any specific token
* sha256 sig as base64 field 
* Artist must publish a public key for public verification

### CODE REVIEW

File: EIPS/eip-721.md [hunk 1]
```diff
@@ -444,4 +444,4 @@ XXXXERC721, by William Entriken -- a scalable example implementation
 
 ## Copyright
 
-Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).
+Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).
```
Comment (claytantor) [inline EIPS/eip-721.md:447]: This didn't change anything. Please don't make me redo this entire PR because of this mistake.

**Comment by claytantor** [comment]
I am assuming there isn't a problem with my commit on the build. I pulled and merged master back prior.

```
clay@clay-deeporb-20:~/data/github.com/claytantor/EIPs$ git checkout master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
clay@clay-deeporb-20:~/data/github.com/claytantor/EIPs$ git pull
Already up to date.
```

File: EIPS/eip-721.md (hunk 1 above)
Comment (MicahZoltu) [reply EIPS/eip-721.md:447]: You can just remove this file/diff from the PR.

Comment (claytantor) [reply EIPS/eip-721.md]: Done

**Comment by pizzarob** [comment]
Provenance of the NFT can already be proven by looking the transaction rather than the event that is emitted.

If I were to look at the transaction for this so-called "sleep-minted" NFT [here](https://etherscan.io/tx/0x1f2066b161f49b607f78b9e08cee07f01cd6da77d5004c3beff3f865dfd305e5?__cf_chl_captcha_tk__=92b139b3eb512a9b0e449f9cb6ae371a7032fc98-1620774732-0-Ac_e3qBgfjMr2_yfSNBnGYqlV79Dp4khmQAtwA8YIxUBmhXNRqBIExSXF4VKk_VMYgBWdJ3vc63CtfW4RTSOO5GN81Re4iBaHYbT8Re6Hn9y9sd4d-0jvO9Rn-IShxKOGeJ06Vhb165fV1usdRpIpxs4stY1LwGnQjKHqJGZ7plOxxFcfP2qj9QjxWfn4fqs3XHO5cEjcsCwSgmTv_ZCKdRSSeUlOkfWrtUDh1Yradgi9AN7kd_GxC7ovAbCDX5qUzEfN4nm-Yb5d05AcJ499MUhHAvAdL_ui9KHEpNsbCkjxulIyQsoml38FUw_vbiw2thCIN98DzSItgtLvTXAF916XtT7SoRszed2dnsNDlVS31KlBmrzFWTLSmdM13sRWn8A9Msou-9UZEmxnNWyGtUUddWt3Dl24POjYQ0x_EKjDUgWB61BP6SMzWK4OsLgyxhvTvB3Q8GnABVyM9B8fQ0rWfIGvEKAl8b2boKElW_6MJI7URIyUTY6yr68G2BgyknyCMXWGFd_xCxszrxCNtq9frj56JiLg5oDFjkVDu0zDOWVvqgyQVZcQI9q0MllO690l6ogI2h-s0KlbCK4VseboWe5ivr98i-lv3Hthrr8aYwWSK6-0YQAJXkIHIY--XqT2P7bJeVoMBHMqOg5ddMxOu9Ezd78WSLBPyHcL3QQpGxMoicrUaFdymtXahJeZkGtMGT9d7Kcb315MDY0mJeSfcaxFrSbOu9hAY7Kvxmme4Js2ELYWJmy2BAhWe1ODw) I can clearly see that someone who does not own the Beeple wallet address created this NFT which was sent to Beeple's wallet during the minting process, but nowhere does it say that Beeple created the NFT. If platforms determine the creators of NFTs by using the initial `Transfer` event then this is flawed and an issue with the platforms not the standard.  Not only that but another way to prove authenticity is through the smart contract address itself. "Sleep minting" is a gimmick and is not an exploit at all.

**Comment by claytantor** [comment]
> That being said I do find this proposal fascinating, but the premise that the "hacker" tricked the blockchain is incorrect. The real issue is that NFT platforms misinterpret the initial `Transfer` event to mean something it does not.

This the best rebuttal to the proposal I have heard, and I agree that a signature *is not necessary* to prove provenance if someone were willing to track down and verify the originating wallet address and do the forensics of comparing it to other minted addresses by the creator. Makes sense. But...

Having a signature in the metadata payload would 100% solid cryptographic approach to watermarking the metadata, and would make every piece of metadata verifiable of minting progeny in the same way that a JWT signature is verifiable. This means artist directories could publish public keys, and DAPs could cryptographically verify any NFT at the time of purchase just by using the getURI call. That seems meaningful to me. In a sense, it says "all assets in this metadata are claimed to be originated by the author at the time of minting whose signature is verifiable"     

I can understand if that isn't meaningful enough to make it part of the standard though, so I won't act like it is.

File: EIPS/eip-3549.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
+---
+eip: 3549
```
Comment (lightclient) [inline EIPS/eip-3549.md:2]:
```suggestion
eip: 3542
```

Comment (lightclient) [reply EIPS/eip-3549.md:2]: Please update the filename to `eip-3541.md` as well.

File: EIPS/eip-3549.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+title: ERC-721 Metadata Signature Standard
+author: Clay Graham <claytantor@gmail.com>
```
Comment (lightclient) [inline EIPS/eip-3549.md:4]:
```suggestion
author: Clay Graham (@claytantor)
```
We require at least one author list a github handle so they can be tagged for notifications.

File: EIPS/eip-3549.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+created: 2020-04-26
+requires: 165
+---
+
+## Simple Summary
+Add signing to the metadata standard and required fields.
+
+## Abstract
+
+Recently Tim Schneider, from Artnet posted a great story about how a whitehat hacker managed to trick the blockchain into forging tokens that would forge the Provenance of the originator of the NFT to look as though it was the originating artist.
```
Comment (lightclient) [inline EIPS/eip-3549.md:17]: Please move this anecdote to the motivation.

File: EIPS/eip-3549.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
+---
+eip: 3549
+title: ERC-721 Metadata Signature Standard
+author: Clay Graham <claytantor@gmail.com>
+type: Standards Track
+category: ERC
+status: Proposed
+created: 2020-04-26
+requires: 165
```
Comment (lightclient) [inline EIPS/eip-3549.md:9]:
```suggestion
requires: 165, 721
```
Is there any reason this wouldn't require EIP-721?

File: EIPS/eip-3549.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+    /// @notice A distinct Uniform Resource Identifier (URI) for a given asset.
+    /// @dev Throws if `_tokenId` is not a valid NFT. URIs are defined in RFC
+    ///  3986. The URI may point to a JSON file that conforms to the "ERC721
+    ///  Metadata JSON Schema".
+    function tokenURI(uint256 _tokenId) external view returns (string);
+
+}
+```
+
+This is the "ERC721 Metadata JSON Schema" referenced above.
```
Comment (lightclient) [inline EIPS/eip-3549.md:78]:
```suggestion
This is the "ERC-721 Metadata JSON Schema" referenced above.
```

File: EIPS/eip-3549.md [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+
+0xcert ERC721 -- a reference implementation
+
+- MIT licensed, so you can freely use it for your projects
+- Includes test cases
+- Active bug bounty, you will be paid if you find errors
+
+
+
+## References
```
Comment (lightclient) [inline EIPS/eip-3549.md:167]: Please remove this section. It is not valid per EIP-1.

File: EIPS/eip-721.md [hunk 8]
```diff
@@ -1,447 +0,0 @@
----
```
Comment (lightclient) [inline EIPS/eip-721.md:1]: Please revert the deletion of this file.

**Review by lightclient — CHANGES_REQUESTED** [review]

**Pull request closed without merging**